### ISSUE TITLE
Panic in cached appveyor build

### ISSUE THREAD

**Issue by phil-opp** [nabijaczleweli/cargo-update#98]
Hi, I'm using cargo-update on appveyor. I have the problem that a panic occurs when using the appveyor cache. The error message is:

```
cargo install-update -a
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 3, kind: NotFound, message: "The system cannot find the path specified." }', libcore\result.rs:1009:5
stack backtrace:
   0: git_filter_source_repo
   1: git_filter_source_repo
   2: git_filter_source_repo
   3: git_filter_source_repo
   4: git_filter_source_repo
   5: git_filter_source_repo
   6: git_filter_source_repo
   7: git_filter_source_repo
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: git_filter_source_repo
  14: git_filter_source_repo
  15: git_filter_source_repo
  16: <unknown>
  17: git_odb_object_size
  18: BaseThreadInitThunk
  19: RtlUserThreadStart
```

I'm caching these directories between builds:

```yaml
cache:
  - '%USERPROFILE%\.cargo\bin'
  - '%USERPROFILE%\.cargo\.crates.toml'
  - target
```

In an uncached build, it works without problems (but takes much longer).

**Comment by phil-opp** [comment]
I were able to reproduce it locally by creating a `~/.cargo` directory with only the `bin` folder and the `.crates.toml` file. The full backtrace is:

<details>

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', libcore/result.rs:1009:5
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:227
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:477
   5: std::panicking::continue_panic_fmt
             at libstd/panicking.rs:391
   6: rust_begin_unwind
             at libstd/panicking.rs:326
   7: core::panicking::panic_fmt
             at libcore/panicking.rs:77
   8: core::result::unwrap_failed
             at libcore/macros.rs:26
   9: <core::result::Result<T, E>>::unwrap
             at libcore/result.rs:808
  10: cargo_update::ops::get_index_path
             at src/ops/mod.rs:652
  11: cargo_install_update::actual_main
             at src/main.rs:61
  12: cargo_install_update::main
             at src/main.rs:21
  13: std::rt::lang_start::{{closure}}
             at libstd/rt.rs:74
  14: std::panicking::try::do_call
             at libstd/rt.rs:59
             at libstd/panicking.rs:310
  15: __rust_maybe_catch_panic
             at libpanic_unwind/lib.rs:103
  16: std::rt::lang_start_internal
             at libstd/panicking.rs:289
             at libstd/panic.rs:392
             at libstd/rt.rs:58
  17: std::rt::lang_start
             at libstd/rt.rs:74
  18: main
  19: __libc_start_main
  20: _start
```

</details><br>


It seems to fail on this `unwrap`: https://github.com/nabijaczleweli/cargo-update/blob/de29530d8bd610f9abd818833438b7979c83641f/src/ops/mod.rs#L651-L653

I guess this means that I have to cache the registry index too.

**Comment by phil-opp** [comment]
Oh, and I'm no longer using cargo-update in my appveyor script, so this issue isn't important for me anymore. Feel free to close this issue (I just left it open in case you want to investigate this panic). Thanks for your time!

**Comment by nabijaczleweli** [comment]
Ah, okay. Well, in that case, you wouldn't (shouldn't) cache the registry, but you'd need to run *some*thing that'd download it (`cargo search "<sth>"` prolly).
I'll make a better error message for this, thanks for reporting!

**Comment by nabijaczleweli** [comment]
Released in [`v1.7.0`](https://github.com/nabijaczleweli/cargo-update/releases/tag/v1.7.0).
